testbed/tests/widgets/test_canvas.py

```diff
@@ -628,6 +629,19 @@ async def test_transforms(canvas, probe):
     assert_reference(probe, "transforms", threshold=0.02)
 
 
+@pytest.mark.xfail(
+    condition=os.environ.get("RUNNING_IN_CI") != "true"
+    and (
+        getattr(
+            importlib.import_module("tests_backend.widgets.canvas").CanvasProbe,
+            "write_text_xfails_local",
+            False,
+        )
+        if importlib.util.find_spec("tests_backend.widgets.canvas")
+        else False
+    ),
+    reason="may fail outside of a GitHub runner environment on this platform",
+)
```

Comment:
This seems a lot more complicated than it needs to be. Canvas tests are known to be fragile on *every* platform. If we're going to mark tests as XFails, I don't think there's a need to do anything more complex than:
```suggestion
    condition=os.environ.get("RUNNING_IN_CI") != "true"
    ),
    reason="Canvas tests are unstable outside of CI. Manual inspection may be required.",
)
```